docs/docs/ref-03-component-specs.md

```diff
@@ -37,6 +37,10 @@ object getInitialState()
 
 Invoked once before the component is mounted. The return value will be used as the initial value of `this.state`.
 
+> Note:
+>
+> This method is not available on ES6 `class` components that extend `React.Component`. It may be removed entirely in a future version of React. For more information, please read our documentation about [ES6 classes](/react/docs/reusable-components.html#es6-classes).
```

Comment:
Let's drop the "may be removed" bit for now. I don't think we've ever talked about that. In a sense if we remove createClass then yes it would have to be removed but for the other methods we have that caveat (replaceState, isMounted), there is an alternative. There is no alternative to getInitialState for createClass.